### PR TITLE
如何通过树莓派的深度学习轻松检测对象

### LINKED ISSUE #3772: 如何通过树莓派的深度学习轻松检测对象
* 原文链接：[How to easily Detect Objects with Deep Learning on Raspberry Pi](https://medium.com/nanonets/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi-225f29635c74)
  * Markdown文件：[文件地址](https://github.com/xitu/gold-miner/blob/master/TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md)
  * PR 地址：#3891
  * 文章分类：**后端**
  * **注意：文件位置在 `TODO1` 文件夹中，不是之前的 `TODO` 了。**
  ----
  * 翻译时间：7 天
  * 校对时间：2 天
  * 翻译积分：7 分
  * 校对积分：2 分
  * [积分有什么用](https://github.com/xitu/gold-miner/wiki)

### PR DESCRIPTION
**Pull request by Starriers**
译文翻译完成，resolve #3772

### CODE REVIEW

**Comment by luochen1992** [comment]
校对认领

**Comment by fanyijihua** [comment]
@luochen1992 好的呢 :beer:

**Comment by jasonxia23** [comment]
认领校对
@fanyijihua

**Comment by fanyijihua** [comment]
@jasonxia23 妥妥哒 :beers:

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 1]
```diff
@@ -141,61 +141,61 @@ for (i<0; i<NoOfCells; i=i+1):
...
 ```
 
-YOLO in <20 lines of code, explained.
+YOLO 在 <20 行代码中的解释。
 
-### How do we build a Deep Learning model for Object Detection?
+### 我们如何构建用于对象检测的深度学习模型？
 
-#### The workflow for Deep Learning has 6 Primary Steps Broken into 3 Parts
+#### 深度学习的工作流的 6 个主要步骤将分成 3 个阶段
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 『深度学习的工作流的』=>『深度学习工作流的』

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 2]
```diff
@@ -141,61 +141,61 @@ for (i<0; i<NoOfCells; i=i+1):
...
 
 ![](https://cdn-images-1.medium.com/max/800/1*hUOIe8skkgMQx68-279z_A.jpeg)
 
 * * *
 
-### Phase 1 — Gather Training Data
+### 阶段 1 —— 收集训练数据
 
-#### **Step 1. Collect Images (at least 100 per Object):**
+#### **第 1 步 收集图像（每个对象至少有 100 个）：**
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 『100 个』=>『100 张图像』

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 3]
```diff
@@ -141,61 +141,61 @@ for (i<0; i<NoOfCells; i=i+1):
...
 
-For this task, you probably need a few 100 Images per Object. Try to capture data as close to the data you’re going to finally make predictions on.
+在这个任务中，每个对象需要几个 100 张图像。尝试将数据捕获到您最终要对其进行预测的数据上。
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 『几个 100 张图像』=>『几百张图像』

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 4]
```diff
@@ -141,61 +141,61 @@ for (i<0; i<NoOfCells; i=i+1):
...
 * * *
 
-### Phase 2 — Training a Model on a GPU Machine
+### 阶段 2 — 在 GPU 机器上训练模型
 
-#### **Step 3. Finding a Pretrained Model for Transfer Learning:**
+#### **第 3 步 寻找可以迁移学习的预训练模型**
 
-You can read more about this at [medium.com/nanonets/nanonets-how-to-use-deep-learning-when-you-have-limited-data-f68c0b512cab](http://medium.com/nanonets/nanonets-how-to-use-deep-learning-when-you-have-limited-data-f68c0b512cab). You need a pretrained model so you can reduce the amount of data required to train. Without it, you might need a few 100k images to train the model.
+您可以在 [medium.com/nanonets/nanonets-how-to-use-deep-learning-when-you-have-limited-data-f68c0b512cab](http://medium.com/nanonets/nanonets-how-to-use-deep-learning-when-you-have-limited-data-f68c0b512cab) 中阅读到更多有关这方面的信息。您需要一个预训练模型，这样您就可以减少训练所需的数据量。没有它，您可能需要几个 10 万张的图像来训练模型。
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 『几个 10 万张』=>『几十万张』

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 5]
```diff
@@ -2,77 +2,77 @@
...
 ![](https://cdn-images-1.medium.com/max/800/1*YJbdykJRHFlzlIXWwn0nIA.gif)
 
-Detecting Vehicles on the Road of Mumbai.
+在孟买的路上检测车辆。
 
-### Why Object Detection?, Why Raspberry Pi?
+### 为什么要检测对象？为什么使用树莓派？
 
-The raspberry pi is a neat piece of hardware that has captured the hearts of a generation with ~15M devices sold, with hackers building even [cooler projects](http://www.trustedreviews.com/opinion/best-raspberry-pi-projects-pi-3-pi-zero-2949390) on it. Given the popularity of Deep Learning and the [Raspberry Pi Camera](https://www.raspberrypi.org/products/camera-module-v2/) we thought it would be nice if we could detect any object using Deep Learning on the Pi.
+树莓派是一款优秀的硬件，它已经捕获了与售卖 1500 万台设备同时代的人的心，甚至于黑客用其构建了[更酷的项目](http://www.trustedreviews.com/opinion/best-raspberry-pi-projects-pi-3-pi-zero-2949390)。鉴于深度学习和[树莓派相机](https://www.raspberrypi.org/products/camera-module-v2/)的流行，我们认为如果能够通过树莓派进行深度学习来检测任何对象，会是一件非常有意义的事情。
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 『它已经捕获了与售卖 1500 万台设备同时代的人的心』=>『它已经吸引了一代人，并且卖出了 1500 万台设备』

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 6]
```diff
@@ -2,77 +2,77 @@
...
 ![](https://cdn-images-1.medium.com/max/800/1*4tPwx3wG720gOmIOaONOEQ.jpeg)
 
-To mimic human level performance scientists broke down the visual perception task into four different categories.
+为了模仿人类层次的表现水平，科学家将视觉感知任务分解为四个不同的类别。
 
-1.  **Classification**, assigns a label to an entire image
-2.  **Localization**, assigns a bounding box to a particular label
-3.  **Object Detection**, draws multiple bounding boxes in an image
-4.  **Image segmentation**, creates precise segments of where objects lie in an image
+1.  **Classification**，为整个图像指定一个标签
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 『Classification』=>『分类』

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 7]
```diff
@@ -2,77 +2,77 @@
...
+2.  **Localization**，为特定标签指定一个边框
+3.  **对象检测**，在图像中绘制多个边界框
+4.  **图像分割**，创建图像中物体所在位置的精确部分
 
-Object detection has been good enough for a variety of applications (even though image segmentation is a much more precise result, it suffers from the complexity of creating training data. It typically takes a human annotator 12x more time to segment an image than draw bounding boxes; this is more anecdotal and lacks a source). Also, after detecting objects, it is separately possible to segment the object from the bounding box.
+对于各种应用来说，对象检测已经足够好了（即使图像分割结果更为精确，但它受到创建训练数据的复杂性影响。它通常需要 12 倍以上的人类注射器，而不是绘制边界框；这是更多的轶事，但缺乏一个来源）。而且在检测对象之后，可以单独从边界框中分割对象。
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 它的意思是，我理解的是，对于一个人类标注者而言，分割图像所花的时间比绘制边界框要多12倍。
注射器什么的，不存在。
希望译者重新翻译一下。

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 8]
```diff
@@ -303,47 +303,47 @@ cd models/research/protoc object_detection/protos/*.proto --python_out=.
...
+我们为您**自动化训练最好的模型**。为了实现这个，我们运行一组具有不同参数的模型，来为您的数据选择最佳模型。
 
-#### 3. No Need for expensive Hardware and GPUs
+#### 3. 不需要昂贵的硬件和 GPU
 
-NanoNets is **entirely in the cloud** and runs without using any of your hardware. Which makes it much easier to use.
+NanoNets **完全在云端运行**而且无需任何硬件。这使得它更容易使用。
 
-#### 4. Great for Mobile devices like the Raspberry Pi
+#### 4. 适合想树莓派这样的移动设备
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 『想』=>『像』

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 9]
```diff
@@ -303,47 +303,47 @@ cd models/research/protoc object_detection/protos/*.proto --python_out=.
...
 
-Since devices like the Raspberry Pi and mobile phones were not built to run complex compute heavy tasks, you can outsource the workload to our cloud which does all of the compute for you
+因为像树莓派和手机这样的设备并不是为了运行复杂的计算任务而构建的，所以您可以吧工作量外包给我们的云，它会为您完成所有的计算
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 『吧』=>『把』

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 10]
```diff
@@ -373,100 +373,100 @@ im.save("image2.jpg")
...
 
 ```
 watch -n 100 python ./code/model-state.py
 ```
 
-#### Step 9: Make Prediction
+#### 第 9 步:预测
 
-Once the model is trained. You can make predictions using the model
+一旦模型被训练，您就可以使用来进行预测
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 『模型被训练』=>『模型训练好了』

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 11]
```diff
@@ -303,47 +303,47 @@ cd models/research/protoc object_detection/protos/*.proto --python_out=.
...
 python ObjectDetectionPredict.py --model data/0/quantized_graph.pb --labels data/label_map.pbtxt --images /data/image1.jpg /data/image2.jpg
 ```
 
 * * *
 
-### Performance Benchmarks on Raspberry Pi
+### 树莓派的性能基准测试
 
-The Raspberry Pi has constraints on both Memory and Compute (a version of Tensorflow Compatible with the Raspberry Pi GPU is still not available). Therefore, it is important to benchmark how much time do each of the models take to make a prediction on a new image.
+树莓派对内存和计算都有限制（与树莓派 GPU 兼容的 Tensorflow 版本仍然不可用）。因此，对基准测试来说，每个模型需要多少是时间才能对新图像进行预测非常重要。
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 『多少是时间』=>『多少时间』

**Comment by leviding** [comment]
@jasonxia23 记得来校对哈

**Comment by leviding** [comment]
@jasonxia23 抓紧校对哈
@Starriers 可以修改啦

**Comment by leviding** [comment]
@Starriers 修改吧

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 12]
```diff
@@ -269,23 +269,23 @@ camera.capture('image1.jpg')
...
+捕获新图像的代码。
 
-#### Step 6. Predicting a new Image
+#### 第 6 步，预测新图像
 
-**Download Model**
+**下载模型**
 
-Once your done training the model you can download it on to your pi. To export the model run:
+一旦你完成了模型的训练，你就可以把它下载到你的圆周率上了。要导出模型运行：
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 『你的圆周率上』=>『你的树莓派上』

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md [hunk 13]
```diff
@@ -303,47 +303,47 @@ cd models/research/protoc object_detection/protos/*.proto --python_out=.
...
 
 * * *
 
 ![](https://cdn-images-1.medium.com/max/800/1*m5grJCpQ6Dk6Ee-JEBIPPg.jpeg)
 
-#### We at NanoNets have a goal of making working with Deep Learning super easy. Object Detection is a major focus area for us and we have made a workflow that solves a lot of the challenges of implementing Deep Learning models.
+#### 我们在 NanoNets 的目标是使深度学习工作更加简单。对象检测是我们关注的一个主要领域，我们已经制定了一个工作流来解决实现深度学习模型的许多挑战。
 
-### How NanoNets make the Process Easier:
+### NanoNets 如何是过程更简单：
```
Comment (luochen1992) [inline TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 『如何是过程更简单』=>『如何使过程更简单』

**Comment by leviding** [comment]
@Starriers 抓紧修改发布了

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md (hunk 5 above)
Comment (Starriers) [reply TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 我觉得 这个 是不是应该强调 售卖 时间 是一个时代的事情

File: TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md (hunk 7 above)
Comment (Starriers) [reply TODO1/how-to-easily-detect-objects-with-deep-learning-on-raspberrypi.md]: 有道理

**Comment by leviding** [comment]
@Starriers 已经 merge 啦~ 快快麻溜发布到[掘金](https://juejin.im)然后给我发下链接，方便及时添加积分哟。

掘金翻译计划有自己的知乎专栏，你也可以投稿哈，推荐使用一个[好用的插件](https://chrome.google.com/webstore/detail/minimalist-markdown-edito/pghodfjepegmciihfhdipmimghiakcjf?hl=zh-CN)。
专栏地址：https://zhuanlan.zhihu.com/juejinfanyi

**Comment by Starriers** [comment]
@luochen1992 @laiyun90 @leviding   校对修改完成，感谢 你们的 校对，辛苦了
拖了 半个月 才弄  我表示 很抱歉 
不好意思了。

**Comment by Starriers** [comment]
已经发布  ：https://juejin.im/post/5b1ba938518825137661af46

**Comment by jasonxia23** [comment]
我也要表示抱歉了，妥了这么久，今天准备校对发现已经合并了
